components/script/script_thread.rs

```diff
@@ -1090,6 +1101,24 @@ impl ScriptThread {
         panic!("Page rect message sent to nonexistent pipeline");
     }
 
+    fn handle_set_scroll_state(&self, id: PipelineId, scroll_state: &Point2D<f32>) {
+        let context = self.browsing_context.get();
+        if let Some(context) = context {
+            if let Some(inner_context) = context.find(id) {
+                let window = inner_context.active_window();
+                window.update_viewport_for_scroll(-scroll_state.x, -scroll_state.y);
+                return
+            }
+        }
+
+        let mut loads = self.incomplete_loads.borrow_mut();
+        if let Some(ref mut load) = loads.iter_mut().find(|load| load.pipeline_id == id) {
+            // FIXME(pcwalton): Do something here?
```

Comment:
I think an incomplete load won't ever be able to get in a state where it receives a set scroll state message, so this could possibly be removed @jdm ?